core/common/src/main/java/alluxio/conf/NestedAlluxioConfiguration.java

```diff
@@ -0,0 +1,178 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.conf;
+
+import com.google.common.collect.ImmutableMap;
+
+import java.time.Duration;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import javax.annotation.concurrent.ThreadSafe;
+
+/**
+ * Configuration that wraps around another configuration
+ * Priority for the value of a property follows:
+ * 1. outer configuration
+ * 2. inner configuration
+ */
+@ThreadSafe
+public class NestedAlluxioConfiguration implements AlluxioConfiguration {
```

Comment:
I'd suggest overlay configuration as the name

```suggestion
public class OverlayConfiguration implements AlluxioConfiguration {
```